source/server/admin/config_dump_handler.cc

```diff
@@ -143,14 +156,23 @@ Http::Code ConfigDumpHandler::handlerConfigDump(absl::string_view url,
   const auto resource = resourceParam(query_params);
   const auto mask = maskParam(query_params);
   const bool include_eds = shouldIncludeEdsInDump(query_params);
+  Matchers::StringMatcherPtr name_matcher;
+  TRY_ASSERT_MAIN_THREAD
+  name_matcher = buildNameMatcher(query_params);
+  END_TRY
+  catch (std::exception& e) {
```

Comment:
With RE2 can this actually throw std::exception or just EnvoyException? Even better, do we even need to use exception handling here? Could we just create a version of the builder that doesn't throw?

Reply:
Sure, I can move this into the builder, and have the builder return StatusOr<StringMatcher>?

Reply:
Yes please, thanks.

Reply:
Done.